racket/collects/openssl/mzssl.rkt

```diff
@@ -646,15 +648,36 @@ TO DO:
                     1))))
           ssl-listener pathname))
 
-(define (ssl-load-private-key! ssl-context-or-listener pathname
+(define (ssl-load-private-key! ssl-context-or-listener path-or-data
                                [rsa? #t] [asn1? #f])
-  (ssl-load-...
-   'ssl-load-private-key!
-   (lambda (ctx path)
-     ((if rsa? SSL_CTX_use_RSAPrivateKey_file SSL_CTX_use_PrivateKey_file)
-      ctx path
-      (if asn1? SSL_FILETYPE_ASN1 SSL_FILETYPE_PEM)))
-   ssl-context-or-listener pathname))
+  (if (and (pair? path-or-data) (eq? (car path-or-data) 'data))
+      (let* ([data (cadr path-or-data)]
+             [ctx (get-context/listener 'ssl-load-private-key! ssl-context-or-listener
+                                        #:need-unsealed? #t)]
+             [bio (BIO_new_mem_buf data (bytes-length data))])
+        (with-failure (lambda () (BIO_free bio))
+          (when asn1?
+            ; TODO: we can probably use d2i_PrivateKey and d2i_RSAPrivateKey to support this if we want
+            (error 'ssl-load-private-key "Cannot load ASN.1 keyfile from bytes at the moment. Must load from file path."))
```

Comment:
This message should follow the [error message conventions](https://docs.racket-lang.org/reference/exns.html#%28part._err-msg-conventions%29) (though obviously it should *not* print the private key!).